language/compiler/ir-to-bytecode/syntax/src/lexer.rs

```diff
@@ -182,114 +94,23 @@ impl<'a> fmt::Display for Token<'a> {
     }
 }
 
-pub struct LexerBuilder {
-    regex_vec: Vec<regex::Regex>,
+pub struct Lexer<'input> {
+    text: &'input str,
+    consumed: usize,
+    previous_end: usize,
+    pub token: (usize, Token<'input>, usize),
```

Comment:
can we make this private?

Reply:
Yes, we can. (At first I thought you meant the Lexer struct, and no, that one must be public. But, that prompted me to realize that the syntax/src/lib.rs file can use the lexer module as a private module.)